src/common/disk_util.rs

```diff
@@ -286,7 +286,7 @@ impl<'a> PartitionIterator<'a> {
                     self.part_idx = 4;
                     self.get_extended_partition()
                 }
-                PartitionType::Fat | PartitionType::Linux => {
+                PartitionType::Fat | PartitionType::Linux | PartitionType::GPT => {
```

Comment:
Why this wasn't needed before? Is it because previously we'd only iterate over partitions of balenaOS versions using MBR partitions? Please consider explaining the "why" of this change in the commit message.

Reply:
Indeed, updated in https://github.com/balena-os/takeover/pull/66/commits/3f5f2dab381e922a441b7ad61de480a4e69a191d

Reply:
Fine to mark this one as resolved or should I add more to the commit message/comments?

Reply:
It's fine, thanks!